pkg/controller/operators/olm/operator.go

```diff
@@ -402,6 +402,18 @@ func (a *Operator) handleClusterServiceVersionDeletion(obj interface{}) {
 			a.csvQueueSet.Requeue(clusterServiceVersion.GetName(), namespace)
 		}
 	}
+
+	if err := a.checkReplacementsAndUpdateStatus(clusterServiceVersion); err == nil {
```

Comment:
Maybe a safer approach here is to delete `APIService`s in `Owned` if they have an owner label referencing this CSV.

Reply:
Ok updated. I guess this is safe since the deletion of a replaced CSV happens after the new CSV is successful.